https://github.com/hyperrealm/libconfig/pull/260
From b90c45a18110fcca415d00a98ff79c908c42544b Mon Sep 17 00:00:00 2001
From: Sergei Trofimovich <slyich@gmail.com>
Date: Thu, 1 May 2025 20:58:26 +0100
Subject: [PATCH] libconfig_format_bin: fix on 32-bit systems

Before the change conversion of bin produced wrong
leading zeros and failed test son `i686-linux` as:

    [TEST] BinaryAndHex
    files "temp.cfg" and "./testdata/binhex.cfg" differ starting at line 5, char 12
    tests.c:55: failed assert: ("temp.cfg") ["temp.cfg"] ==txtfile (output_file) ["./testdata/binhex.cfg"]
    [FAIL] BinaryAndHex

Closes: https://github.com/hyperrealm/libconfig/issues/259
---
 lib/util.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/util.c b/lib/util.c
index 5455cac..cf73766 100644
--- a/lib/util.c
+++ b/lib/util.c
@@ -286,7 +286,7 @@ void libconfig_format_double(double val, int precision, int sci_ok, char *buf,
 /* ------------------------------------------------------------------------- */
 
 #ifdef __GNUC__
-#define clzl(x) __builtin_clzl(x)
+#define clzl(x) __builtin_clzll(x)
 #else
 static int clzl(int64_t val)
 {
